sorting/quicksort3.cpp

```diff
@@ -0,0 +1,68 @@
+#include <iostream>
+#include <vector>
+#include <cstdlib>
+
+using std::vector;
+using std::swap;
+
+vector<int> quick_sort_partition3(vector<int> &a, int l, int r) {
```

Comment:
```
sorting/quicksort3.cpp:8:  Is this a non-const reference? If so, make const or use a pointer: vector<int> &a  [runtime/references] [2]
sorting/quicksort3.cpp:31:  Is this a non-const reference? If so, make const or use a pointer: vector<int> &a  [runtime/references] [2]
sorting/quicksort3.cpp:44:  Is this a non-const reference? If so, make const or use a pointer: vector<int> &a  [runtime/references] [2]
sorting/quicksort3.cpp:49:  Consider using rand_r(...) instead of rand(...) for improved thread safety.  [runtime/threadsafe_fn] [2]
```
Please use  const or use a pointer; a non-const reference is detected.